src/cascadia/TerminalApp/Pane.h

```diff
@@ -19,140 +19,169 @@
+    //   that is.
+    // - This Pane's control might not currently be focused, if the tab itself is
+    //   not currently focused.
+    // Return Value:
+    // - nullptr if we're a leaf and unfocused, or no children were marked
+    //   `_lastActive`, else returns this
+    virtual std::shared_ptr<LeafPane> FindActivePane() = 0;
 
-    void UpdateSettings(const winrt::Microsoft::Terminal::Settings::TerminalSettings& settings,
-                        const GUID& profile);
-    void ResizeContent(const winrt::Windows::Foundation::Size& newSize);
-    bool ResizePane(const winrt::TerminalApp::Direction& direction);
-    bool NavigateFocus(const winrt::TerminalApp::Direction& direction);
+    // Method Description:
+    // - Invokes the given action on each descendant leaf pane, which may be just this pane if 
+    //   it is a leaf.
+    // Arguments:
+    // - action: function to invoke on leaves. Parameters:
+    //      * pane: the leaf pane on which the action is invoked
+    // Return Value:
+    // - <none>
+    virtual void PropagateToLeaves(std::function<void(LeafPane&)> action) = 0;
 
-    bool CanSplit(winrt::TerminalApp::SplitState splitType);
-    std::pair<std::shared_ptr<Pane>, std::shared_ptr<Pane>> Split(winrt::TerminalApp::SplitState splitType,
-                                                                  const GUID& profile,
-                                                                  const winrt::Microsoft::Terminal::TerminalControl::TermControl& control);
+    // Method Description:
+    // - Invokes the given action on each descendant leaf pane that touches the given edge of this
+    //   pane. In case this is a leaf pane, it is assumed that it touches every edge of itself.
+    // Arguments:
+    // - action: function to invoke on leaves. Parameters:
+    //      * pane: the leaf pane on which the action is invoked
+    // Return Value:
+    // - <none>
+    virtual void PropagateToLeavesOnEdge(const winrt::TerminalApp::Direction& edge,
+                                         std::function<void(LeafPane&)> action) = 0;
+
+    // Method Description:
+    // - Attempts to update the settings of this pane or any children of this pane.
+    //   * If this pane is a leaf, and our profile guid matches the parameter, then
+    //     we'll apply the new settings to our control.
+    //   * If we're not a leaf, we'll recurse on our children.
+    // Arguments:
+    // - settings: The new TerminalSettings to apply to any matching controls
+    // - profile: The GUID of the profile these settings should apply to.
+    // Return Value:
+    // - <none>
+    virtual void UpdateSettings(const winrt::Microsoft::Terminal::Settings::TerminalSettings& settings,
+                                const GUID& profile) = 0;
 
-    void Close();
+    virtual void ResizeContent(const winrt::Windows::Foundation::Size& newSize) = 0;
+    virtual void Relayout() = 0;

```

Comment:
I feel like `Relayout` should just be named `Layout`.

Reply:
That's #3181

Reply:
OK. Sorry, I didn't read 3181 before this one. Lots of PRs. :)